classes/submission/PKPAuthorDAO.inc.php

```diff
@@ -297,25 +298,23 @@ function deleteObject($author) {
 
 	/**
 	 * Delete an author by ID.
-	 * @param $authorId int
-	 * @param $submissionId int optional
+	 * @param $authorId int Author ID
+	 * @param $submissionId int Optional submission ID.
 	 */
 	function deleteById($authorId, $submissionId = null) {
 		$params = array((int) $authorId);
 		if ($submissionId) $params[] = (int) $submissionId;
-		$returner = $this->update(
+		$this->update(
 			'DELETE FROM authors WHERE author_id = ?' .
 			($submissionId?' AND submission_id = ?':''),
 			$params
 		);
-		if ($returner) $this->update('DELETE FROM author_settings WHERE author_id = ?', array((int) $authorId));
-
-		return $returner;
```

Comment:
Did you check if there were places in the code using this function waiting for a return value?